core/src/main/java/io/fabric8/maven/core/util/KubernetesResourceUtil.java

```diff
@@ -402,7 +404,17 @@ public static String extractContainerName(MavenProject project, ImageConfigurati
     private static String extractImageUser(String image, MavenProject project) {
         ImageName name = new ImageName(image);
         String imageUser = name.getUser();
-        return imageUser != null ? imageUser : project.getGroupId();
+        String projectGroupId = project.getGroupId();
+        if(imageUser != null) {
+            return imageUser;
+        } else {
+            if(projectGroupId == null || projectGroupId.matches(CONTAINER_NAME_REGEX)) {
+                return projectGroupId;
+            }
+            else {
+                return projectGroupId.replaceAll("[^a-zA-Z0-9-]", "");
```

Comment:
This replacement still does not ensure that a name does not start or end with a `-`. Ok, a groupId propably never starts with a minus, but maybe it could end with one ?

I would add something like `.replace("^-*(.*)-*$","\\1")` ...

Reply:
Sorry, please correct me if I'm wrong. I tested my regex with groupId ending with a `-`, I think it handles. Plus how does your regex trims `.` from groupId? Does `replace` work with regex? I tried your regex with `replaceAll` but it seems to be replacing everything with 1

Reply:
@rohanKanojia I mean to add the regexp I suggest in _addition_ to yours of course (and it was not tested, therefore that _something like_ ...)

```
projectGroupId.replaceAll("[^a-zA-Z0-9-]", "").replaceFirst("^-*(.*)-*$","\\1")
```

The idea is to remove leading and trailing `-` from the name as these are not allowed.